### DISCUSSION TITLE
请问能够默认包含头文件和 自定义宏 进行检测, 不需要通过 #include 包含这个头文件

### DISCUSSION THREAD

**Discussion by baron-z** [neoclide/coc.nvim#4967] [Q&A]
1. 我在使用 coc.nvim 中添加 linux kernel 代码
在 kernel 中会在 Makefile 定义 ``` __KERNEL__ ```这个宏, 但是我在使用  coc.nvim 的时候没法默认添加这个宏

```
#define __KERNEL__   // 需要在这里显示定义
#include <linux/module.h>
......

 MODULE_AUTHOR("baron"); //这里不会报错
```

```
// #define __KERNEL__   // 需要在这里显示定义, 如果这里没有定义
#include <linux/module.h>
......

>> 80 MODULE_AUTHOR("baron"); //这里就会报错
```

因此 coc.nvim 没有办法默认配置``` __KERNEL__ ``` 这个宏适用于所有代码.

2. 同样的 linux kernel 中的 .config 会自动编译成为 include/generated/autoconf.h
有办法默认使用这个头文件嘛, 而不是使用 #inclue <include/generated/autoconf.h> 显示包含.

这两个配置我在网上没有找到对应的解决方案, 
希望能得到解惑, 
十分感谢.

**Comment by fannheyward** [comment]
coc.nvim 是一个 LSP client，这些头文件查找、宏定义检测等，都是 language server 处理的，client 只是把错误显示。
